File: Criacionais/Singleton.py
class Singleton:
    _instance = None  # Armazena a única instância da classe

    def __new__(cls):
        # Se _instance é None, cria uma nova instância
        if cls._instance is None:
            cls._instance = super(Singleton, cls).__new__(cls)
        return cls._instance  # Retorna a instância existente ou recém-criada

# Testando o Singleton
s1 = Singleton()
s2 = Singleton()
print(s1 == s2)  # True, pois ambos apontam para a mesma instância (Singleton)

File: Criacionais/Singleton.java
package Criacionais;

public class Singleton {
    private static Singleton instance = null; // Armazena a única instância da classe

    private Singleton() {
        // Construtor privado impede a criação de instâncias fora da classe
    }

    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton(); // Cria a instância se for a primeira vez
        }
        return instance; // Retorna a instância única
    }

    public void showMessage() {
        System.out.println("Singleton Pattern Example");
    }

    public static void main(String[] args) {
        Singleton singleton1 = Singleton.getInstance();
        Singleton singleton2 = Singleton.getInstance();
        
        singleton1.showMessage();
        System.out.println(singleton1 == singleton2); // True, ambas são a mesma instância
    }
}
